File: NavitiaSDKSandbox/NavitiaSDKSandbox/Places/EndpointRequestBuilderPlaces.swift
//
// Created by Chakkra CHAK on 20/04/2017.
// Copyright (c) 2017 Chakkra CHAK. All rights reserved.
//

import Foundation

open class EndpointRequestBuilderPlaces: BaseNavitiaRequestBuilder {
    // <Generated>
    public required init(navitiaConfiguration: NavitiaConfiguration) {
        super.init(navitiaConfiguration: navitiaConfiguration, resourceUri: "/places")
    }
    // </ Generated>

    // <Generated>
    public func get(callback: @escaping (EndpointResponsePlaces) -> (Void)) {
        return self.genericGet(processResponseHandler: { (data: Data) -> EndpointResponsePlaces in
            let json: [String: AnyObject] = try JSONSerialization.jsonObject(with: data, options: .allowFragments) as! [String: AnyObject]

            return EndpointResponsePlaces(json: json)!
        }, callback: callback)
    }
    // </ Generated>

    // <Generated>
    open func withQ(_ q: String) -> EndpointRequestBuilderPlaces {
        addQueryParameter(key: "q", value: q)
        return self
    }
    // </ Generated>

    // <Generated>
    open func withCount(_ count: Int) -> EndpointRequestBuilderPlaces {
        addQueryParameter(key: "count", value: count.description)
        return self
    }
    // </ Generated>
}

File: android/NavitiaSDKSandbox/src/main/java/org/kisio/labs/NavitiaSDKSandbox/Places/EndpointRequestBuilderPlaces.java
package org.kisio.labs.NavitiaSDKSandbox.Places;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;
import org.json.simple.JSONObject;
import org.json.simple.parser.JSONParser;
import org.json.simple.parser.ParseException;
import org.kisio.labs.NavitiaSDKSandbox.BaseNavitiaRequestBuilder;
import org.kisio.labs.NavitiaSDKSandbox.NavitiaConfiguration;

import java.io.IOException;

public class EndpointRequestBuilderPlaces extends BaseNavitiaRequestBuilder {
    public EndpointRequestBuilderPlaces(NavitiaConfiguration navitiaConfiguration) {
        super(navitiaConfiguration, "/places");
    }

    public EndpointRequestBuilderPlaces withQ(String q) {
        this.addQueryParameter("q", q);
        return this;
    }

    public EndpointRequestBuilderPlaces withCount(int count) {
        this.addQueryParameter("count", String.valueOf(count));
        return this;
    }
}
